deepchem/models/dft/nnxc.py

```diff
@@ -0,0 +1,152 @@
+from abc import abstractproperty, abstractmethod
+from typing import Union
+try:
+    from dqc.xc.base_xc import BaseXC
+    from dqc.utils.datastruct import ValGrad, SpinParam
+    from dqc.api.getxc import get_xc
+    import torch
+except ModuleNotFoundError:
+    raise ModuleNotFoundError("This layer requires dqc and torch")
+
+
+class BaseNNXC(BaseXC, torch.nn.Module):
+    """
+    Base class for the NNLDA and HybridXC classes.
+    """
+
+    @abstractproperty
+    def family(self) -> int:
+        pass
+
+    @abstractmethod
+    def get_edensityxc(
+            self, densinfo: Union[ValGrad, SpinParam[ValGrad]]) -> torch.Tensor:
+        pass
+
+
+class NNLDA(BaseNNXC):
+    """
+    Neural network xc functional of LDA (only receives the density as input)
+    """
+
+    def __init__(self,
+                 nnmodel: torch.nn.Module,
+                 ninpmode: int = 1,
+                 outmultmode: int = 1):
+        super().__init__()
+        """
+        Parameters
+        ----------
+        nnmodel: torch.nn.Module
+            Neural network for xc functional
+        ninpmode: int
+            The mode to decide the transformation of the density to NN input.
+        outmultmode: int
```

Comment:
What are the valid values for the output? What does each value mean?

Reply:
Currently we only support one mode, so I have removed it. Will add it back in the next iterations.